internal/service/rds/cluster.go

```diff
@@ -1180,6 +1183,13 @@ func resourceClusterUpdate(ctx context.Context, d *schema.ResourceData, meta int
 			input.EngineVersion = aws.String(d.Get("engine_version").(string))
 		}
 
+		// This can happen when updates are deferred (apply_immediately = false), and
+		// multiple applies occur before the maintenance window. In this case,
+		// continue sending the desired engine_version as part of the modify request.
+		if d.Get("engine_version").(string) != d.Get("engine_version_actual").(string) {
```

Comment:
Is there anything to be gained from checking whether `apply_immediately` _is_ `false` here?

Reply:
Good question, edge cases are tough to reason about here. One potential case could be an out of band deferred engine_version upgrade.

- Config has `apply_immediately = true`, `engine_version = 1.0.0`
- Major `engine_version` upgrade (2.0.0) is done in the console, but deferred
- On the next read, `engine_version = 2.0.0` (from pending updates), `engine_version_actual = 1.0.0`

If the user then updates their config `engine_version = "2.0.0"` (now matches refreshed state) and runs an `apply`, the current implementation would hit this conditional and set the `EngineVersion` input to `2.0.0`. If we added an `if !d.Get("apply_immediately")` condition, neither the `d.HasChange("engine_version")` or this condition would return true, leaving `EngineVersion` unset in the ModifyDBCluster request.